src/SFML/Graphics/Shape.cpp

```diff
@@ -143,6 +145,44 @@ FloatRect Shape::getGlobalBounds() const
     return getTransform().transformRect(getLocalBounds());
 }
 
+Vector2f Shape::getGeometricCenter() const
+{
+    const auto count = getPointCount();
+
+    switch (count)
+    {
+        case 2:
+            return (getPoint(0) + getPoint(1)) / 2.f;
+        case 1:
+            return getPoint(0);
+        case 0:
+            sf::err() << "Shape::getGeometricCenter() called on empty Shape";
+            std::terminate();
+        default:
+            const auto offset   = getPoint(0);
+            const auto offsetX2 = 2.f * offset;
+            Vector2f   centroid;
+            float      twicearea{};
+            float      f;
+
+            auto p2 = getPoint(count - 1);
+            for (std::size_t i = 0; i < count; ++i)
+            {
+                const auto p1 = getPoint(i);
+
+                f = (p1.x - offset.x) * (p2.y - offset.y) - (p2.x - offset.x) * (p1.y - offset.y);
+                twicearea += f;
+                centroid += ((p1 + p2) - offsetX2) * f;
```

Comment:
I don't think offset is needed. Also this can be more readable using `cross` method.

```suggestion
                const float product = p2.cross(p1);
                twicearea += product;
                centroid += (p1 + p2) * product;
```

Reply:
@kimci86 Without the factor `f` and `offsetX2` the centroid is placed very far away. I'm sure I misunderstood how to make the change. Can you please show the whole `default` section of the function?

Reply:
I imagine this should work:
```c++
default:
{
    Vector2f centroid;
    float    twicearea{};

    auto p2 = getPoint(count - 1);
    for (std::size_t i = 0; i < count; ++i)
    {
        const auto p1 = getPoint(i);

        const float product = p2.cross(p1);
        twicearea += product;
        centroid += (p1 + p2) * product;

        p2 = p1;
    }

    return centroid / 3.f / twicearea;
}
```

Reply:
@kimci86 Yes, very nice! Updated the patch now! Thanks!